cli/src/commands/init.js

```diff
@@ -183,7 +186,42 @@ const handler = async ({ force, name, cwd, lib }) => {
         })
     }
 
-    const entrypoint = lib ? 'src/index.jsx' : 'src/App.jsx'
+    if (typeScript) {
+        // copy tsconfig
+        reporter.info('Copying tsconfig')
+        fs.copyFileSync(paths.initTSConfig, paths.tsConfig)
+
+        reporter.info('install TypeScript as a dev dependency')
+        // ToDO: restrict the major version of TS we install
+        await exec({
+            cmd: 'yarn',
+            args: ['add', 'typescript', '--dev'],
+            cwd: paths.base,
+        })
+
+        // Todo: check why d2 start is renaming TS files to JS
+
+        // ToDO: make custom eslint config part of the template (and copy it)0
+        // similar to: https://github.com/dhis2/data-exchange-app/pull/79/files#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5
+        // install dependencies needed for eslint
+        // "@typescript-eslint/eslint-plugin"
+        // "@typescript-eslint/parser"
+
+        // ToDO: install any other TS dependencies needed
```

Comment:
Would this include `types/react[-dom]`?

Reply:
Maybe even `@types/jest`

<img width="1013" alt="Screenshot 2024-08-05 at 5 31 30 PM" src="https://github.com/user-attachments/assets/2c03b5bd-4b96-4f02-9aca-1774c06870a6">